hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +85,957 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
+                                    access_location::host,
+                                    access_mode::read);
+    for (unsigned int idx = 0; idx < num_group; ++idx)
+        {
+        // collect the initial velocities of the embedded particles
+        const unsigned int particle_index = h_embed_group.data[idx];
+        h_initial_vel.data[idx] = h_velocity.data[particle_index];
+        }
+    }
+
+void mpcd::CollisionMethod::thermalizeConstituentParticles(uint64_t timestep)
+    {
+    // zero accumulators
+    m_linmom_accum.zeroFill();
+    m_angmom_accum.zeroFill();
+
+    Scalar T_set = (*m_T)(timestep);
+
+    ArrayHandle<Scalar4> h_velocity(m_pdata->getVelocities(),
+                                    access_location::host,
+                                    access_mode::readwrite);
+    ArrayHandle<Scalar4> h_alt_vel(m_pdata->getAltVelocities(),
+                                   access_location::host,
+                                   access_mode::overwrite);
+    ArrayHandle<unsigned int> h_tag(m_pdata->getTags(), access_location::host, access_mode::read);
+    ArrayHandle<Scalar4> h_postype(m_pdata->getPositions(),
+                                   access_location::host,
+                                   access_mode::read);
+    ArrayHandle<int3> h_image(m_pdata->getImages(), access_location::host, access_mode::read);
+    const BoxDim& global_box = m_pdata->getGlobalBox();
+
+    ArrayHandle<Scalar3> h_linmom_accum(m_linmom_accum,
+                                        access_location::host,
+                                        access_mode::readwrite);
+    ArrayHandle<Scalar3> h_angmom_accum(m_angmom_accum,
+                                        access_location::host,
+                                        access_mode::readwrite);
+    ArrayHandle<unsigned int> h_lookup_center(m_rigid_bodies->getLookupCenters(),
+                                              access_location::host,
+                                              access_mode::read);
+    unsigned int num_total = m_pdata->getN();
+    uint16_t seed = m_sysdef->getSeed();
+
+    // get random velocities and accumulate the resulting change in momentum
+    for (unsigned int idx = 0; idx < num_total; ++idx)
+        {
+        // get the index from the particle and check if it is a constituent
+        const unsigned int central_idx = h_lookup_center.data[idx];
+        if (central_idx >= MIN_FLOPPY)
+            {
+            continue;
+            }
+        // do not need to thermalize central particle
+        if (idx == central_idx)
+            {
+            continue;
+            }
+
+        Scalar mass_const = h_velocity.data[idx].w;
+        unsigned int tag = h_tag.data[idx];
+        // draw random velocities from normal distribution
+        hoomd::RandomGenerator rng(
+            hoomd::Seed(hoomd::RNGIdentifier::CollisionMethod, timestep, seed),
+            hoomd::Counter(tag));
```

Comment:
I'm not sure in the current form: are there other places where `CollisionMethod` or its children uses a `RandomGenerator`? If yes, we may need to mix an extra integer into the counter and/or add a seed.

Reply:
No:
> anywhere you have added a RandomGenerator
Like I said above, the second integer defaults to zero. The children were the only place this was used before, so they have the second integer 0 by default. Start from 1 for any additional instances of the `RandomGenerator` that you **added**.

Reply:
okay